### ISSUE TITLE
Does qlib support distributed training on multiple machines?

### ISSUE THREAD

**Issue by pingsutw** [microsoft/qlib#217]
I didn't see any docs about workflow with distributed training on multiple machines.
Are you going to support this feature?

**Comment by you-n-g** [comment]
@pingsutw 
If you are asking about parallel training multiple individual models on multiple machines. 
```
Currently, qlib provides the APIs which user can define individual tasks.
Such API will be helpful to parallel training for multiple models.   User can dispatch these tasks to different machines and gather the results after traning. 

Training on multiple machines is related to the underlying infrastructure.

We havel an internal version for task management based on mongodb. But we don't think is genral enough. So it is not released now.
```

If you are asking about distributed training of a single model. 
We think it will be supported in model level instead of framework level of Qlib.

Thanks.

**Comment by pingsutw** [comment]
@you-n-g Thanks for your reply.

> If you are asking about distributed training of a single model.
> We think it will be supported in model level instead of framework level of Qlib.

I know that it's related to the implementation of the model.
And sorry that I didn't clarify my question.

Let's say if I want to run this [workflow](https://github.com/microsoft/qlib/blob/main/examples/benchmarks/LSTM/workflow_config_lstm_Alpha158.yaml), and this model (qlib.contrib.model.pytorch_lstm_ts) is implement by qlib.
Could I train this model on multiple machines using `ps-worker` strategy? 

Currently, I can use `qrun` to run the workflow in a server. Is possible to use it run on Kubeflow or other ML platforms.
In other words, Could I run the model in `qlib` in a distributed environment?

**Comment by you-n-g** [comment]
@pingsutw 
I think qlib will not prevent you from using it in a distributed environment.
Qlib is a framework and defines the interface each modules. It doesn't restrict the implementation of each module. 

For example, you can implement a model in any way.  The only thing Qlib cares about is that you  implemented the right interface https://github.com/microsoft/qlib/blob/main/qlib/model/base.py#L21

**Comment by pingsutw** [comment]
Ok, thanks for your information. I will try to implement a model by using this interface.
